packages/core/strapi/lib/services/event-hub.js

```diff
@@ -1,16 +1,30 @@
+'use strict';
+
 /**
  * The event hub is Strapi's event control center.
  */
+module.exports = function createEventHub() {
+  // Store of subscribers that will be called when an event is emitted
+  const subscribers = [];
 
-'use strict';
+  return {
+    emit(eventName, ...args) {
```

Comment:
We could make this async so we can wait for all subscriber to have executer that could come in handy